client/flags/flags.go

```diff
@@ -112,6 +112,7 @@ func AddTxFlagsToCmd(cmd *cobra.Command) {
 
 // AddPaginationFlagsToCmd adds common pagination flags to cmd
 func AddPaginationFlagsToCmd(cmd *cobra.Command, query string) {
+	cmd.Flags().Uint64(FlagPage, 1, fmt.Sprintf("pagination page of %s to query for. This is being deprecated, prefer to use --offset", query))
```

Comment:
I don't understand why we're marking this as deprecated yet it's going into `master`? Why not just remove it outright?

Reply:
The main argument for keeping it was to not break backwards-compatibility.

Reply:
Stargate will be breaking in many ways though, the legacy API being an exception. Is there a strong rationale for keeping this?

Reply:
> Stargate will be breaking in many ways though, the legacy API being an exception. Is there a strong rationale for keeping this?

Yes, there are still a couple of CLI queries which require `page` flag, https://github.com/cosmos/cosmos-sdk/blob/e6a56226a88ec4e8761b1388a3a666af627d9e2e/x/auth/client/cli/query.go#L168-L171, https://github.com/cosmos/cosmos-sdk/blob/e6a56226a88ec4e8761b1388a3a666af627d9e2e/x/gov/client/cli/query.go#L287-L291

Updating them would require an update to https://github.com/cosmos/cosmos-sdk/blob/e6a56226a88ec4e8761b1388a3a666af627d9e2e/x/auth/client/query.go#L23 or we should convert `offset` to `page` in CLI. As we are thinking to keep things backward compatible as much as possible, bringing back `page` flag was a fix this.

Reply:
Also, REST is still using `page` flag, having it enabled on `REST` and removing it from `CLI` didn't sound like a better UX

Reply:
Ok, that's fine and makes sense. However, my point of contention is that I don't think we should mark something as "deprecated" before a major release -- we should opt to remove it entirely or remove the deprecation mark. Maintaining backwards compatibility is not the same thing as deprecation usually. 

However, since we're keeping the legacy API around for a release, we can also keep this around for a release.

Reply:
I don't think we actually need to mark this as deprecated. It's a convenience feature for CLI users and I don't see a problem leaving it. The fact that `offset` and `limit` are used to handle this under the hood doesn't mean everyone should switch over to `offset`. `--page` makes it easy for CLI users.